WordPress/gutenberg
# [RNMobile] Generic Embed Block placeholder UI

###  Gutenberg Mobile https://github.com/wordpress-mobile/gutenberg-mobile/pull/3346

## Description
This PR implements the Generic Embed Block placeholder UI. 

Based on the `embed/edit.js` the placeholder is shown when :

### Embed web

1. There's no preview. 
2. When we can't embed the current URL. 
3. When we have clicked the edit button. 

Supporting code [here.](https://github.com/WordPress/gutenberg/blob/3da717b8d0ac7d7821fc6d0475695ccf3ae2829f/packages/block-library/src/embed/edit.js#L192)

### Embed native

I am assuming that we are currently supporting numbers 1 and 2. I made that assumption because when we are editing the URL ( link ) we will be utilizing the "Bottom Sheet to set a link" approach so I don't see the need for utilizing the behavior tied to number 3. I will look more into the logic later to get a better understanding of it. If you have any thoughts here, I am happy to hear them. 

## How has this been tested?
1. Open the **Inserter.**
2. Add the _Embed_ block to the canvas.
3. Notice the _Embed_ Block **placeholder** UI. 

## Screenshots 

## Android

Light Mode | Dark Mode 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/113960474-2e4ef500-97ea-11eb-9a62-46023f80c0d2.png" width="320"></kbd>|    <kbd><img src="https://user-images.githubusercontent.com/1509205/113960884-c51bb180-97ea-11eb-8178-27c82f65c3e2.png" width="320"></kbd>
   
## iOS

Light Mode | Dark Mode 
--------|-------
 <kbd><img src="https://user-images.githubusercontent.com/1509205/113960645-648c7480-97ea-11eb-910b-8b848ccad2ca.png" width="320"></kbd>    |       <kbd><img src="https://user-images.githubusercontent.com/1509205/113960649-65bda180-97ea-11eb-82b0-a4461c39c950.png" width="320"></kbd>
 
 ## Regression Testing - Web
 
 1. run `npm run wp-env start`
 2. Log into wpadmin.
 3. Add an Embed block. 
 4. Verify that the label of the placeholder says "Embed URL" and not "Embed". 

![web-test](https://user-images.githubusercontent.com/1509205/113972384-29953b80-9800-11eb-8bb7-97fc03fcf43a.png)

## Types of changes

The primary changes made here are: 

1. The label that is utilized the embed block is generated based on the `Platform` in use because the web version requires `URL` to be appended to the embed variant. eg, `Twitter URL`. While the mobile design specs did not specify this behavior. 
2. The mobile **Embed Placeholder** was modified to represent the design specifications in the [attached issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3273). Since the UI of this `Embed Placeholder` is similar to that of the `Media Placeholder` I reused the layout system and styles. I thought about refactoring the styles and views related to the placeholder so they could be reused in more than one place. In this PR, I have duplicate logic (views and styles) for rendering the UI. I choose this approach in case we need to do customizations for the Embed block and it was a faster approach to get the placeholder ready quickly. However, I am not against changing it.  
3. I added _accessibility_ behavior to the **placeholder** UI so that it provides the appropriate role and hint to the accessibility framework of the respective **platform**. I have not verified this as yet with screen readers hence the reason it is not checked below. 

Changes I excluded from this PR: 

1. I did not create any unit tests or e2e tests because I wanted to wait until the block has reached further along. 
2. I did add any `prop` to the `onPress` because I expect the **BottomSheet** PR to attach itself to the **placeholder** and wire up that behavior. Let me know if this needs to be included in this PR.
3. The **placeholder** UI is not in a _function_ i.e the **render** method only supports one UI _state_. However, I was thinking that when the other states such as _loading_, _error_ are introduced the **render** method could be refactored to support this. 

## Checklist:
